src/pages/ProfilesExplorerView/components/SceneExploreDiffFlameGraph/SceneExploreDiffFlameGraph.tsx

```diff
@@ -79,6 +81,51 @@ export class SceneExploreDiffFlameGraph extends SceneObjectBase<SceneExploreDiff
         this.state.presetsPicker.openSelect();
       })
     );
+
+    this._subs.add(
+      this.subscribeToEvent(EventEnableSyncTimeRanges, (event) => {
+        const { enable } = event.payload;
+        const { baselinePanel, comparisonPanel } = this.state;
+
+        comparisonPanel.toggleTimeRangeSync(enable);
+        baselinePanel.toggleTimeRangeSync(enable);
+
+        this.state.presetsPicker.toggle(enable);
+      })
+    );
+
+    this._subs.add(
+      // eslint-disable-next-line sonarjs/cognitive-complexity
```

Comment:
Should we tweak the rule instead of skipping it?

Reply:
I realize I can refactor the code to prevent the complexity.